src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -361,13 +356,7 @@ private Action<TRow> GenerateSetter(IRow input, int index, InternalSchemaDefinit
                     {
                         Ch.Assert(colType.IsBool);
                         Ch.Assert(peek == null);
-                        return CreateConvertingActionSetter<DvBool, bool>(input, index, poke, x => (bool)x);
-                    }
-                    else if (fieldType == typeof(bool?))
-                    {
-                        Ch.Assert(colType.IsBool);
-                        Ch.Assert(peek == null);
-                        return CreateConvertingActionSetter<DvBool, bool?>(input, index, poke, x => (bool?)x);
+                        return CreateConvertingActionSetter<bool, bool>(input, index, poke, x => x);
```

Comment:
>CreateConvertingActionSetter [](start = 31, length = 28)

fall back to // T -> T

Reply:
falling back ...

---
In reply to: [211109169](https://github.com/dotnet/machinelearning/pull/692#discussion_r211109169) [](ancestors = 211109169)